src/app/modules/spcp/spcp.util.ts

```diff
@@ -1,13 +1,40 @@
 import { StatusCodes } from 'http-status-codes'
 
 import { createLoggerWithLabel } from '../../../config/logger'
-import { MapRouteError } from '../../../types'
+import { AuthType, MapRouteError, MyInfoAttribute } from '../../../types'
 import { MissingFeatureError } from '../core/core.errors'
 
 import { CreateRedirectUrlError } from './spcp.errors'
 
 const logger = createLoggerWithLabel(module)
 
+type SpcpLocals =
+  | {
+      uinFin: string
+      hashedFields: Set<MyInfoAttribute>
+    }
+  | { uinFin: string; userInfo: string }
+  | { [key: string]: never } // empty object
```

Comment:
I suppose a literal empty object {} doesn't work here?

Reply:
not recommended, Typescript does not typecheck `{}` types well.